### DISCUSSION TITLE
Time-based schedule

### DISCUSSION THREAD

**Discussion by rsmb7z** [nautechsystems/nautilus_trader#896] [Q&A] [answered]
Hi @cjdsellers @limx0 
Is there way to setup strategy restart at given scheduled time?

**Comment by limx0** [comment, marked as answer]
Hey @rsmb7z, not explicitly in nautilus, this is more something you would do outside of your python process (via cron or the like). If you wanted to keep your state between runs you can use redis as a cache database and persist any strategy data there on shutdown.

Does that suit your use case ?

**Reply by rsmb7z** [reply]
Hi @limx0 
Thanks for the response. The state is not important to me but the reason brought me to this is because I noticed that data subscription is being canceled at mid-night for some reason (Bars not received after), so I thought of restarting strategy which will eventually re-subscribe. To start new process using cron we need to exit current one before that so connection ports are released. I think seperate actor (looping as time observer) shall be able to perform time-based events, if I am thinking on right track?

**Comment by limx0** [comment]
Yeah so I have a feeling that might be the IB midnight reset, the way I handle this is setting a shutdown timer q hour after market close and a cron to restart. 

This has the added benefit of being a good time to upgrade code etc. I'm not sure we want to tackle this inside of nautilus, but happy to brainstorm some potential solutions. 

You're correct it would be trivial to run an actor for time based events, the trouble is the actor runs as somewhat of an isolated instance without the power to affect the rest of the system.

**Reply by rsmb7z** [reply]
The thing is market for Forex (IDEALPRO) and Futures (CME/NYMEX) is open that time, it closes for 15min and one-hour respectively at 17:00 (EST) during which Nautilus/ib_sync work smooth when it is open back. I have raised the issue at ib_insync to see if this can be fixed in there.
